internal/controller/cloud-resources/aws/awsnfsbackupschedule_test.go

```diff
@@ -4,67 +4,94 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/google/uuid"
 	cloudcontrolv1beta1 "github.com/kyma-project/cloud-manager/api/cloud-control/v1beta1"
+	cloudresourcesv1beta1 "github.com/kyma-project/cloud-manager/api/cloud-resources/v1beta1"
 	"github.com/kyma-project/cloud-manager/pkg/kcp/nfsinstance"
 	skrawsnfsvol "github.com/kyma-project/cloud-manager/pkg/skr/awsnfsvolume"
 	"github.com/kyma-project/cloud-manager/pkg/skr/backupschedule"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
-	cloudresourcesv1beta1 "github.com/kyma-project/cloud-manager/api/cloud-resources/v1beta1"
 	. "github.com/kyma-project/cloud-manager/pkg/testinfra/dsl"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// I can not make this work, good luck figuring this one
-var _ = SkipDescribe("Feature: SKR AwsNfsBackupSchedule", func() {
+var _ = Describe("Feature: SKR AwsNfsBackupSchedule", func() {
 
 	const (
-		interval = time.Millisecond * 50
-	)
-	var (
-		timeout = time.Second * 20
+		interval        = time.Millisecond * 50
+		timeout         = time.Second * 20
+		awsAccountId    = "974658265573"
+		toleranceWindow = 120 * time.Second
 	)
-	var now time.Time
-	var skrNfsVolumeName string
-	var skrNfsVolume *cloudresourcesv1beta1.AwsNfsVolume
-	var nfsInstanceName string
-	var nfsInstance *cloudcontrolv1beta1.NfsInstance
-	var awsNfsId string
-	var scope *cloudcontrolv1beta1.Scope
 
-	awsAccountId := "974658265573"
+	It("Scenario: Creates recurring backup schedule with existing backups", func() {
+		var (
+			suffix             string
+			skrNfsVolumeName   string
+			nfsInstanceName    string
+			scheduleName       string
+			existingBackupName string
+			skrNfsVolume       *cloudresourcesv1beta1.AwsNfsVolume
+			nfsInstance        *cloudcontrolv1beta1.NfsInstance
+			scope              *cloudcontrolv1beta1.Scope
+			backupSchedule     *cloudresourcesv1beta1.AwsNfsBackupSchedule
+			existingBackup     *cloudresourcesv1beta1.AwsNfsVolumeBackup
+		)
 
-	BeforeEach(func() {
-		By("Given KCP Scope exists", func() {
-			now = time.Now().UTC()
-			skrNfsVolumeName = uuid.NewString()
-			skrNfsVolume = &cloudresourcesv1beta1.AwsNfsVolume{}
-			nfsInstanceName = uuid.NewString()
-			nfsInstance = &cloudcontrolv1beta1.NfsInstance{}
-			awsNfsId = uuid.NewString()
-			scope = &cloudcontrolv1beta1.Scope{}
+		// This ensures each parallel test run has different resource names
+		suffix = fmt.Sprintf("recurring-%d", GinkgoParallelProcess())
```

Comment:
Usual practice is to hard code unique uuid (use some online tool like https://www.uuidtools.com/v4) into a variable
```suggestion
		name = "6278871d-a536-4702-87a1-ffc21dfc38ad"
```